### PR TITLE
Items: Get All items, Get All sorted items (low to high), Get one item by id , Create (then delete) one item

### PR DESCRIPTION
**Pull request by pshim17**
Get All items && Get All sorted items (low to high)

- Completed the index method in the items_controller
- Tests all passing in Postman:
   - Happy fetch, returns all items in the correct format.
   - Happy path for sort parameter, all items returned and sorted by price in the correct format.
- Added gem "jsonapi-serializer" but serializer was NOT used, can come back later to add.

### CODE REVIEW

**Comment by pshim17** [comment]
Get one item by id

- Completed the show method in the items_controller
- Tests all passing in Postman:
   - happy path, fetch one item by id in the correct format
   - sad path, bad integer id returns 404
   - edge case, string id returns 404

**Comment by pshim17** [comment]
Create (then delete) one item

- Added Create method to the item_controller
- Added Destroy method to the item_controller
- Added routes for post/delete
- Tests passing in Postman:
   - response should be okay to process
   - returns 201 status

File: app/controllers/api/v1/items_controller.rb [hunk 1]
```diff
@@ -1,5 +1,74 @@
 class Api::V1::ItemsController < ApplicationController
   def index
-    
+    if params[:sorted] === 'price'
+      items = Item.all.order("unit_price asc")
```
Comment (Jonathan-Atkins) [inline app/controllers/api/v1/items_controller.rb:4]: did this not need a semicolon before asc to work? Nice work!

File: Gemfile [hunk 2]
```diff
@@ -35,6 +35,8 @@ gem "bootsnap", require: false
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin Ajax possible
 gem "rack-cors"
 
+gem "jsonapi-serializer"
```
Comment (Jonathan-Atkins) [inline Gemfile:38]: glad we are on same page using gem. Last project we hand rolled them- it was good practice but happy to use the gem now.

File: config/routes.rb
Comment (Jonathan-Atkins) [inline config/routes.rb:1]: good REST

**Review by Jonathan-Atkins — APPROVED** [review]

File: app/controllers/api/v1/items_controller.rb [hunk 3]
```diff
@@ -1,5 +1,74 @@
...
+          unit_price: item.unit_price,
+          merchant_id: item.merchant_id
+        }
+      }
+    end
+    render json: { data: itemsFormatted }
+  end
+
+  def show
+    item = Item.find(params[:id])
```
Comment (KMPWeiland) [inline app/controllers/api/v1/items_controller.rb:25]: What happens if Item.find(params[:id]) fails in the show action? MIght be a place to incorporate error handling

**Review by KMPWeiland — APPROVED** [review]
Nice work!
